File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.rs
impl Solution {
    pub fn length_of_longest_substring(s: String) -> i32 {
        let mut str_vec : Vec<char> = Vec::new();
        let parsed_string : Vec<char> = s.chars().collect();
        let mut max : i32 = 0;
        let mut count = 0;
        for i in parsed_string.iter() {
            while str_vec.contains(i) {
                str_vec.remove(0);
            }
            str_vec.push(*i);
            count = str_vec.len() as i32;
            if count > max {
                max = count;
            }
        }
        max
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        set<int> set;
        int ans = 0;
        int maxLength = 0;
        
        for(int i = 0;i < s.size(); i++) {
            while(set.find(s[i]) != set.end()) {
                set.erase(s[ans]);
                ans++;
            }
            set.insert(s[i]);
            maxLength = max(maxLength, i-ans+1);
        }
        
        return maxLength;
    }
};